psi4/driver/driver_util.py

```diff
@@ -282,50 +252,102 @@ def parse_cotton_irreps(irrep: Union[str, int], point_group: str) -> int:
 
 
-def negotiate_derivative_type(ptype: str,
-                              method: str,
-                              user_dertype: Optional[int],
-                              verbose: int = 1,
-                              return_strategy: bool = False,
-                              proc: Optional[Dict] = None) -> Union[int, str]:
+def negotiate_derivative_type(
+    target_dertype: str,
+    method: str,
+    user_dertype: Optional[int],
+    verbose: int = 1,
+    proc: Optional[Dict] = None,
+) -> Union[Tuple[int, int], Tuple[str, str]]:
     r"""Find the best derivative level (0, 1, 2) and strategy (analytic, finite difference)
-    for `method` to achieve `ptype` within constraints of `user_dertype`.
+    for `method` to achieve `target_dertype` within constraints of `user_dertype`.
 
     Parameters
     ----------
-    ptype : {'energy', 'gradient', 'hessian'}
+    target_dertype: {"energy", "gradient", "hessian", "properties"}
         Type of calculation targeted by driver.
     method
         Quantum chemistry method targeted by driver. Should be correct case for procedures lookup.
     user_dertype
-        User input on which derivative level should be employed to achieve `ptype`.
+        User input on which derivative level should be employed to achieve `target_dertype`.
     verbose
         Control amount of output printing.
-    return_strategy
-        See below. Form in which to return negotiated dertype.
     proc
         For testing only! Procedures table to look up `method`. Default is psi4.driver.procedures .
+    managed_keywords
+        NYI Keywords that influence managed methods.
 
     Returns
     -------
-    int : {0, 1, 2}
-        When `return_strategy=False`, highest accessible derivative level
-        for `method` to achieve `ptype` within constraints of `user_dertype`.
-    str : {'analytic', '1_0', '2_1', '2_0'}
-        When `return_strategy=True`, highest accessible derivative strategy
-        for `method` to achieve `ptype` within constraints of `user_dertype`.
+    tuple : (int, int)
+        "second" is highest accessible derivative level for `method` to achieve
+        `target_dertype` "first" within constraints of `user_dertype`. When
+        "first" == "second", analytic is the best strategy, otherwise finite
+        difference of target "first" by means of "second".
 
     Raises
     ------
     ValidationError
-        When input validation fails. When `user_dertype` exceeds `ptype`.
+        When input validation fails. When `user_dertype` exceeds `target_dertype`.
     MissingMethodError
         When `method` is unavailable at all. When `user_dertype` exceeds what available for `method`.
 
     """
-    egh = ['energy', 'gradient', 'hessian']
+    if target_dertype == "properties":
+        if isinstance(method, list):
+            # untested
+            analytic = [highest_analytic_derivative_available(mtd, proc) for mtd in method]
```

Comment:
Do you mean `highest_analytic_properties_available`?

Reply:
quite right, now fixed. `properties("cbs")` has never been attempted, so this is just prep.